modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/cache/CacheIndexesRebuild.java

```diff
@@ -0,0 +1,239 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.commandline.cache;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.apache.ignite.internal.client.GridClient;
+import org.apache.ignite.internal.client.GridClientConfiguration;
+import org.apache.ignite.internal.commandline.AbstractCommand;
+import org.apache.ignite.internal.commandline.Command;
+import org.apache.ignite.internal.commandline.CommandArgIterator;
+import org.apache.ignite.internal.commandline.TaskExecutor;
+import org.apache.ignite.internal.commandline.argument.CommandArgUtils;
+import org.apache.ignite.internal.commandline.cache.argument.IndexRebuildCommandArg;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.internal.visor.cache.index.IndexRebuildTaskArg;
+import org.apache.ignite.internal.visor.cache.index.IndexRebuildTaskRes;
+
+import static org.apache.ignite.internal.commandline.CommandLogger.INDENT;
+import static org.apache.ignite.internal.commandline.cache.argument.IndexRebuildCommandArg.NODE_ID;
+import static org.apache.ignite.internal.commandline.cache.argument.IndexRebuildCommandArg.TARGET;
+
+/**
+ * Cache subcommand that schedules indexes rebuild via the maintenance mode.
+ */
+public class CacheIndexesRebuild extends AbstractCommand<CacheIndexesRebuild.Arguments> {
+    /** Command's parsed arguments. */
+    private Arguments args;
+
+    /** {@inheritDoc} */
+    @Override public void printUsage(Logger logger) {
+        String desc = "Schedules rebuild of the indexes for specified caches.";
+
+        Map<String, String> map = U.newLinkedHashMap(16);
+
+        map.put(NODE_ID.argName(), "(Optional) Specify node for indexes rebuild.");
+        map.put(
+            TARGET.argName(),
+            "Cache name with optionally specified indexes. If indexes are not specified then all indexes of the cache will be scheduled "
+            + "for the rebuild operation."
+        );
+
+        usageCache(
+            logger,
+            CacheSubcommands.INDEX_REBUILD,
+            desc,
+            map,
+            NODE_ID.argName() + " nodeId",
+            TARGET + " cacheName1=index1,...indexN"
```

Comment:
I think it would be better to have two arguments and have them in the following options:
- --cache-names=cache0,cache1[idx0,idx2]
- --group-names=grp0,grp1[idx0,idx2]

Reply:
Why do you need group names? Indexes belongs to caches, not groups

Reply:
Why do you need group names? Indexes belong to caches, not groups

Reply:
Yes, for groups, you can remove the index refinement, the idea is that you need to give the opportunity to rebuild the indexes for all caches in the group.

Reply:
You can do this, just don't specify any caches: "(Optional) Specify node for indexes rebuild."

Reply:
Why would you want to rebuild indexes for all caches in the group? Let's not overcomplicate things